### PR TITLE
add azure_rm_mariadbserver  module

### PR DESCRIPTION
**Pull request by techknowlogick**
##### SUMMARY
azure_rm_mariadb (and other modules), would support the creation of MariaDB servers using azure resource manager

Credit to: @zikalino as this is almost a 1:1 clone of their MySQL work on azure.

##### ISSUE TYPE
- New Module Pull Request

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_mariadbconfiguration.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Zim Kalinowski, (@zikalino)
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_mariadbconfiguration.py]: Pls add yourself here :-)

**Comment by techknowlogick** [comment]
ready for review

CI is failing because the subscription is not registered to use namespace 'Microsoft.DBforMariaDB', to get tests to pass, the above namespace will need to be added to the subscription that is used for tests.

**Comment by zikalino** [comment]
@mattclay could you please add MariaDB to the subscription?

**Comment by mattclay** [comment]
The `Microsoft.DBforMariaDB` provider has been registered. I'll restart the tests.

**Comment by zikalino** [comment]
looks good!
@techknowlogick could you create pr with remaining modules so we can catch 2.8 freeze next week?

**Review by zikalino — APPROVED** [review]

**Comment by techknowlogick** [comment]
@zikalino Will do. Thanks for review & merge 😄

**Pull request merged**